Include/internal/pycore_symtable.h

```diff
@@ -10,7 +10,9 @@ extern "C" {
 
 struct _mod;   // Type defined in pycore_ast.h
 
-typedef enum _block_type { FunctionBlock, ClassBlock, ModuleBlock, AnnotationBlock }
+typedef enum _block_type { FunctionBlock, ClassBlock, ModuleBlock,
```

Comment:
I wonder if it might decrease a possible "overwhelmed by complexity" feeling from future maintainers of this code if we had an explicit comment here clarifying what you've had to clarify to at least one reviewer: that all the `Type*` scopes are really just `FunctionBlock` but with some syntax disallowed; the only functional difference is in issuing some errors in the compiler.

Reply:
Added some comments